File: 01_LOAD_SHAPEFILE.py
# Import the arcpy library
import arcpy

# Define the path to the shapefile (correcting the syntax error)
shapefile_path = r"C:/Users/alvarece/Desktop/MODELAMIENTO Y PROGRAMACION GIS/PRACTICAS/01_CARGA_SHAPEFILES/DATOS/MOROCCO_BOUNDARIES.shp"

# Load the shapefile into a feature class (feature layer)
shapefile_fc = arcpy.management.MakeFeatureLayer(shapefile_path, "shapefile_layer")

# Print information about the shapefile
desc = arcpy.Describe(shapefile_fc)
print(f"Feature Type: {desc.shapeType}")
print(f"Spatial Reference: {desc.spatialReference.name}")

# List the fields in the shapefile
fields = [field.name for field in arcpy.ListFields(shapefile_fc)]
print("Fields in the shapefile:", fields)

File: 01_LOAD_SHAPEFILE/01_LOAD_SHAPEFILE.R
# Check if the sf package is installed, and install it only if it's missing
if (!requireNamespace("sf", quietly = TRUE)) {
  install.packages("sf")
}

# Load the sf library
library(sf)

# Define the path to your shapefile
shapefile_path <- "C:/Users/alvarece/Desktop/MODELAMIENTO Y PROGRAMACION GIS/PRACTICAS/01_CARGA_SHAPEFILES/DATOS/MOROCCO_BOUNDARIES.shp"

# Read the shapefile
shapefile_data <- st_read(shapefile_path)

# Print summary
print(shapefile_data)

# Plot the shapefile
plot(shapefile_data$geometry)

# Get CRS (Coordinate Reference System)
crs_info <- st_crs(shapefile_data)
print(paste("CRS:", crs_info$epsg, "-", crs_info$proj4string))

# Get field names (attribute columns)
fields <- colnames(shapefile_data)
print("Fields in the shapefile:")
print(fields)
